src/Sylius/Component/Taxonomy/Generator/TaxonSlugGenerator.php

```diff
@@ -52,6 +52,10 @@ public function generate($name, $parentId = null)
         $parent = $this->taxonRepository->find($parentId);
         Assert::notNull($parent, sprintf('There is no parent taxon with id %d.', $parentId));
 
+        if ($parentLocale) {
+            $parent->setCurrentLocale($parentLocale);
+        }
+
         return $parent->getSlug().self::SLUG_SEPARATOR.$taxonSlug;
```

Comment:
`$parent->getTranslation($locale)->getSlug().self::SLUG_SEPARATOR.$taxonSlug` will be a better solution. `$parent->getTranslation($locale)` could be even extracted to variable.